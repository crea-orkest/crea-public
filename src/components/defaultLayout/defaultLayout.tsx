import { Footer } from 'components/footer'
import { Navigation } from 'components/navigation'
import classNames from 'classnames'
import styles from './styles.module.scss'

export function DefaultLayout({ children }: { children: React.ReactNode }) {
  return (
    <div className={classNames(styles.main)}>
      <Navigation />
      <div>{children}</div>
      <Footer />
    </div>
  )
}
