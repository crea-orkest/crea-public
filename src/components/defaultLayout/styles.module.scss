.main {
  display: grid;
  grid-template-rows: max-content 100vh max-content;
  min-height: 100vh;
  align-items: start;
}
