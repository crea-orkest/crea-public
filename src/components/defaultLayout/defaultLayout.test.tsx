import { DefaultLayout } from './defaultLayout'
import React from 'react'
import { render } from '@testing-library/react'

jest.mock('../navigation', () => {
  const originalModule = jest.requireActual('../navigation')
  return {
    __esModule: true,
    ...originalModule,
    Navigation: jest.fn(() => <nav>[Navigation]</nav>),
  }
})

jest.mock('../footer', () => {
  const originalModule = jest.requireActual('../footer')
  return {
    __esModule: true,
    ...originalModule,
    Footer: jest.fn(() => <footer>[Footer]</footer>),
  }
})

describe('Default layout', () => {
  it('shows all the data', () => {
    const { container } = render(
      <DefaultLayout>
        <main>main content</main>
      </DefaultLayout>
    )
    expect(container).toMatchSnapshot()
  })
})
