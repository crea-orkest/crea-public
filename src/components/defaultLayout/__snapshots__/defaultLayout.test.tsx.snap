// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default layout shows all the data 1`] = `
<div>
  <div
    class="main"
  >
    <nav>
      [Navigation]
    </nav>
    <div>
      <main>
        main content
      </main>
    </div>
    <footer>
      [Footer]
    </footer>
  </div>
</div>
`;
